content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/system-extensions.md

```diff
@@ -0,0 +1,48 @@
+* Call `DependencyNotificationBus::Events::OnCompositionChanged` or `DependencyNotificationBus::Events::OnCompositionRegionChanged` whenever any data within the gradient changes. This notifies everything "above" the gradient that it will need to refresh itself.
+
+Ideally, new gradient components would also support the following:
+
+* Use the **Shape** components to define how the gradient data should map into world space.
+* Support the **Gradient Transform Modifier** component for performing arbitrary transforms to the gradient data.
+* By convention, all gradients currently listen to `OnEntityVisibilityChanged` and use that to enable or disable the gradient data in the Editor. This makes it easy to turn gradients on and off while authoring, though it's important to recognize that they will still be active at runtime regardless of the authoring visibility setting.
+
+## Terrain Components
+
+Each of the terrain components can be replaced with new components that meet the same requirements. For example, a new component could be created that streams down real-world satellite height data as a replacement for the **Terrain Height Gradient List**. It's also possible to combine components together, such as creating a simplified Terrain Spawner component that implements the functionality of the **Terrain Layer Spawner**, **Terrain Height Gradient List**, and **Terrain Surface Gradient List** on one single component.
+
+Any replacement components need to support the following:
+
+* Implement the request bus that matches the component being replaced (`TerrainSpawnerRequestBus`, ``TerrainAreaHeightRequestBus`, TerrainAreaSurfaceRequestBus`, etc).
+* Declare that it implements each service that matches the component being replaced (`TerrainAreaService`, `TerrainHeightProviderService`, `TerrainSurfaceProviderService`, etc) in `GetProvidedServices`.
+* Support thread safety.
+* Call `OnCompositionChanged` or `OnCompositionRegionChanged` whenever any data within the component changes.
+* Call the appropriate APIs on the `TerrainSystemServiceRequestBus` to register/unregister/refresh a terrain region.
+
+## Terrain physics integration
+
+The terrain physics support is provided through a divided component system. One half is the `TerrainPhysicsCollider`, which knows about the generic `TerrainDataRequestBus` for terrain data and the generic `HeightfieldProvider` buses for physics communication, and the other half is the `PhysXHeightfieldCollider`, which knows about PhysX and heightfields, but not about terrain. To integrate terrain with a new physics system, a new `*HeightfieldCollider` component would need to be implemented to replace the PhysX version that works with the `HeightfieldProvider` buses and with the new physics system.
```

Comment:
Does the * before HeightfieldCollider signify something, or is this a typo?

Reply:
It's a placeholder mark, because the replacement would be "BulletHeightfieldCollider", "JoltHeightfieldCollider", "PhysX51HeightfieldCollider", or whatever. I can see where it should probably get removed though, it might just cause more confusion than value to have it there.

Reply:
Removed the *.